proxy-robots-module/src/main/java/orwell/proxy/config/Configuration.java

```diff
@@ -35,19 +37,13 @@ private void populate() {
 
             switch (enumConfigFileType) {
                 case RESOURCE:
-                    final InputStream xml = getClass().getResourceAsStream(filePath);
-                    this.configuration = (ConfigModel) unmarshaller.unmarshal(xml);
-                    isPopulated = true;
-                    logback.info("Configuration loaded from resource file");
+                    onResource(unmarshaller);
                     break;
                 case FILE:
-                    final File file = new File(filePath);
-                    this.configuration = (ConfigModel) unmarshaller.unmarshal(file);
-                    isPopulated = true;
-                    logback.info("Configuration loaded from external file");
+                    onFile(unmarshaller);
                     break;
                 default:
-                    logback.error("Config file type of " + enumConfigFileType + " not handled");
+                    onDefault();
```

Comment:
And now you might think that I would suggest something like populateFromNotHandled, but no. I would suggest to create a dedicated object for file, resource, url (and whatnot) thus invalidating the previous two comments. This would allow you to get rid of enumConfigFileType, that is so hard to track and use polymorphism instead.